app/validators/flood_risk_engine/grid_reference_validator.rb

```diff
@@ -38,7 +39,7 @@ def grid_reference_pattern
     end
 
     def two_letters
-      "[A-Z]{2}"
```

Comment:
seems incredibly verbose, to create a simple regex, and then not that clear what the final regex actually is